### PR TITLE
Ensure that all the examples work for Standard.Base.Data

### LINKED ISSUE #1674: Add `Table.from_csv : Text | File -> Table`
### General Summary

What it says on the tin!

### Motivation

Better working with in-memory CSV data.

### PR DESCRIPTION
**Pull request by iamrecursion**
### Pull Request Description
Does what it says on the tin.

Closes #1674.

### Important Notes
Consolidated `parse` and `parse_format` for the time types.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: distribution/std-lib/Standard/src/Base/Data/Vector.enso [hunk 1]
```diff
@@ -119,7 +117,8 @@ type Vector
 
        > Example
          Checking the length of a vector.
-             [1, 2, 3, 4].length == 4
+
+             [1, 2, 3, 4].length
```
Comment (4e6) [inline distribution/std-lib/Standard/src/Base/Data/Vector.enso:121]: AFAIK, for the `[ ]` syntax we need a `Standard.Base` import.

Here and below.

Comment (iamrecursion) [reply distribution/std-lib/Standard/src/Base/Data/Vector.enso:121]: The assumption for all examples is that `from Standard.Base import all` is present. I should document this.

File: distribution/std-lib/Standard/src/Base/Data/Range.enso [hunk 2]
```diff
@@ -15,23 +15,26 @@ type Range
...
     length = this.end - this.start
 
     ## Checks if this range is empty.
 
        > Example
-             0.up_to 0 . is_empty == True
-             0.up_to 100 . is_empty == False
+         Checking if the range from 0 to 100 is empty.
+
+             0.up_to 100 . is_empty
```
[the remark this answers is not in the document]
Comment (iamrecursion) [reply distribution/std-lib/Standard/src/Base/Data/Range.enso:28]: You don't need to. The `.up_to` method is exported from the prelude and you don't need a type imported to use methods defined on it.

File: distribution/std-lib/Standard/src/Base/Data/Text/Extensions.enso [hunk 3]
```diff
@@ -85,12 +90,14 @@ Text.characters =
...
      Split the comma-separated text into a vector of items.
+
          "ham,eggs,cheese,tomatoes".split ","
 
    > Example
      Split the string on whitespace into a vector of items.
+
          "ham eggs cheese tomatoes".split Split_Kind.Whitespace
-Text.split : Split_Kind -> Vector.Vector
+Text.split : Split_Kind -> Vector.Vector Text
```
[the remark this answers is not in the document]
Comment (iamrecursion) [reply distribution/std-lib/Standard/src/Base/Data/Text/Extensions.enso:100]: Not sure, but I'm not the first to do it!

File: distribution/std-lib/Standard/src/Base/Data/Vector.enso (hunk 1 above)
Comment (iamrecursion) [reply distribution/std-lib/Standard/src/Base/Data/Vector.enso:121]: I've now documented this.

**Review by 4e6 — APPROVED** [review]
Epic!

**Pull request merged**